core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -717,17 +717,16 @@ class KafkaApis(val requestChannel: RequestChannel,
       val internalTopicsAllowed = request.header.clientId == AdminUtils.ADMIN_CLIENT_ID
 
       // call the replica manager to append messages to the replicas
-      replicaManager.appendRecords(
+      replicaManager.handleProduceAppend(
         timeout = produceRequest.timeout.toLong,
         requiredAcks = produceRequest.acks,
         internalTopicsAllowed = internalTopicsAllowed,
         origin = AppendOrigin.CLIENT,
```

Comment:
nit: I wonder if we should remove the `origin` parameter from `handleProduceAppend` as it should always come from a client by definition.

Reply:
that's fair. I can do that.